test-data/pybind11_mypy_demo/stubgen-include-docs/pybind11_mypy_demo/basics.pyi

```diff
@@ -0,0 +1,134 @@
+from typing import ClassVar
+
+from typing import overload
+PI: float
+
+class Point:
+    class AngleUnit:
+        __members__: ClassVar[dict] = ...  # read-only
+        __entries: ClassVar[dict] = ...
+        degree: ClassVar[Point.AngleUnit] = ...
```

Comment:
Can we simplify this fixture? Or do we need everything from it?

Reply:
Not everything is needed for the docstring test, but the C++ source (`test-data/pybind11_mypy_demo/src/main.cpp`) was already used for testing stubgenc.

Modifying it to test both the default stubgenc behaviour and the optional docstrings seemed beneficial, since the original stubgenc test didn't contain any docstrings, so it did not test what happens if there were any. And now it serves also as an input for the optional docstring generation.

I agree that this (expected) test output is a bit more bloated than necessary, but also the test scenario `misc/test-stubgenc.sh` is IMHO easier to understand and maintain than a version that cleans up and compiles different C++ code for different test cases.